pkg/kinflate/util/labelsandannotations.go

```diff
@@ -0,0 +1,80 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+// MapTransformationOptions contains a map string->string and path configs
+// The map will be applied to the fields specified in path configs.
+type MapTransformationOptions struct {
+	m           map[string]string
+	pathConfigs []PathConfig
+}
+
+var _ Transformer = &MapTransformationOptions{}
+
+// CompleteForLabels fills up the MapTransformationOptions for labels transformation.
+func (o *MapTransformationOptions) CompleteForLabels(m map[string]string, pathConfigs []PathConfig) {
```

Comment:
sorry, previous review somehow dropped a comment here.

This is basically a double setter with a hidden default, and raises alarms for me.

Please drop this, in favor of an object that is immutable behind the Transformer interface:

```
func NewDefaultingLabelsMapTransformer(m map[string]string) (Transformer, err) {
  return NewMapTransformer(DefaultLabelsPathConfigs, m)
}

func NewDefaultingAnnotationsMapTransformer(m map[string]string) (Transformer, err) {
  return NewMapTransformer(DefaultAnnotationsPathConfigs, m)
}

func NewMapTransformer(pc []PathConfig, m map[string]string) (Transformer, err) {
  if (m == nil or is otherwise bad in some way) {
   return nil, errors.New("bad map")
  }
  if (pc == nil or is otherwise bad in some way) {
   return nil, errors.New("bad pathconfigs")
  }
  return &MapTransformer{pc, m}, nil
}
```